src/kits/tracker/ContainerWindow.cpp

```diff
@@ -1663,20 +1629,15 @@ BContainerWindow::MessageReceived(BMessage* message)
 							&& TargetModel()->IsDirectory()
 							&& !PoseView()->IsFilePanel()
 							&& !PoseView()->IsDesktopWindow()) {
-							BRect rect(Bounds());
-							rect.top = KeyMenuBar()->Bounds().Height() + 1;
-							rect.bottom = rect.top
-								+ BNavigator::CalcNavigatorHeight();
-							fNavigator = new BNavigator(TargetModel(), rect);
+							fNavigator = new BNavigator(TargetModel());
+							fPoseContainer->GridLayout()->AddView(fNavigator, 0, 0, 2);
 							fNavigator->Hide();
-							AddChild(fNavigator);
 							SetPathWatchingEnabled(settings.ShowNavigator()
 								|| settings.ShowFullPathInTitleBar());
 						}
 
 						if (!settings.SingleWindowBrowse()
-							&& dynamic_cast<BDeskWindow*>(this) == NULL
-							&& TargetModel()->IsDesktop()) {
+							&& fIsDeskWindow && TargetModel()->IsDesktop()) {
 							// close the "Desktop" window, but not the Desktop
```

Comment:
You revert the logic of the if (fIsDeskWindow vs. == NULL). I would assume that this code should prevent windows to open that show the desktop besides the actual desktop.